### PR TITLE
New pack beep is back!

### PR DESCRIPTION
**Pull request by sirugh**
This adds the beep back. I may have been trying to get too clever with the audio react component but I just wanted to see what happened if I created one. Also works if we just add an audio el to the index but then it wouldn't be manageable by react.

https://github.com/aeosynth/draft/issues/97

### CODE REVIEW

File: public/src/components/audio.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import _ from '../../lib/utils'
+import App from '../app'
+let d = React.DOM
+
+export default React.createClass({
+  render() {
```
Comment (sirugh) [inline public/src/components/audio.js]: Can I do this: 

```
export default React.createClass({
  let el = d.audio({ id : 'beep', src : '/media/beep.wav'})
  beep() : {
    this.getDOMNode().play()
  },
  render() {
    return el
  }
})`

and then trigger it in cards.js by importing Audio and calling `Audio.beep()`

Comment (sirugh) [inline public/src/components/audio.js]: http://stackoverflow.com/questions/25986439/reactdom-getdomnode-instead-of-jquery

**Comment by sirugh** [comment]
Hey @aeosynth is there anything I can do for this or the other PR's to get them merged?

File: public/src/components/audio.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import _ from '../../lib/utils'
+import App from '../app'
```
Comment (aeosynth) [inline public/src/components/audio.js]: you're not using these imports

File: public/src/cards.js [hunk 3]
```diff
@@ -95,6 +95,8 @@ let events = {
       pack[name] = 1
     }
     App.update()
+    if (document.hidden && App.state.beep)
```
Comment (aeosynth) [inline public/src/cards.js]: beeping is binary; a user could have the tab open but be focused on another window. there needs to be a setting to support beeping only while the tab is hidden

**Comment by aeosynth** [comment]
i want the game component to only worry about rendering; cards.js should encapsulate beeping. stick it in index

**Comment by aeosynth** [comment]
remove the beep file from `public` if you're putting it in `media`

File: public/src/components/audio.js (hunk 1 above)
Comment (aeosynth) [inline public/src/components/audio.js]: this code isn't generic enough to justify being a separate component

File: public/src/cards.js (hunk 3 above)
Comment (sirugh) [inline public/src/cards.js]: Regarding the focusing on another window, could [this](https://developer.mozilla.org/en-US/docs/Web/API/document.hasFocus) work:

`if (App.state.beep && (document.hidden || !document.hasFocus))`

Comment (sirugh) [inline public/src/cards.js]: Or is `document.hidden` true even when the tab is visible but not focused?

Comment (aeosynth) [inline public/src/cards.js]: woah, i did not know about `document.hasFocus`. regardless, the user could simply be looking away from his screen

Comment (sirugh) [inline public/src/cards.js]: We can just do away with the document checking. If they have beep on pack enabled, beep it!

**Comment by aeosynth** [comment]
add the audio to the index instead of going through react

**Comment by aeosynth** [comment]
you can ping me when you make the changes

**Comment by sirugh** [comment]
Hmm.. I tried 

```
<body>
  <script>
    System
    .import('./src/init')
    .catch(function(err) { console.log(err.message) })
  </script>
  <audio id="beep" src="/media/beep.wav"></audio>
</body>
```

but the `audio` el is getting removed on render.

**Comment by sirugh** [comment]
I'm not clear on what's happening, but it seems like React is overwriting the body. I tried this:

```
<script>
    System
    .import('./src/init')
    .then(function () {
      document.body.insertAdjacentHTML('afterend', '<audio id="beep" src="/media/beep.wav"></audio>');
    })
    .catch(function(err) { console.log(err.message) })
  </script>
```
and it appended a second body

screen shot 2015-02-04 at 12 02 50 am

**Comment by aeosynth** [comment]
oh yeah, react does that - http://facebook.github.io/react/docs/top-level-api.html#react.render

> *Note*:

> React.render() replaces the contents of the container node you pass in. In the future, it may be possible to insert a component to an existing DOM node without overwriting the existing children.

add the audio element in game.js directly

**Comment by sirugh** [comment]
@aeosynth Addressed your comments.

File: public/src/cards.js [hunk 4]
```diff
@@ -95,6 +95,9 @@ let events = {
       pack[name] = 1
     }
     App.update()
+    if (App.state.beep){
+      self.beep.play()
+    }
```
Comment (aeosynth) [inline public/src/cards.js]: don't use braces for single statement blocks

Comment (aeosynth) [inline public/src/cards.js]: self is not defined here. use `getElementById`

File: public/index.html [hunk 5]
```diff
@@ -10,6 +10,7 @@
   <script src="lib/react.js"></script>
   <script src="lib/engine.io.js"></script>
   <script src="lib/traceur.js"></script>
+  <audio id="beep" src="/media/beep.wav"></audio>
```
Comment (aeosynth) [inline public/index.html]: this is not needed

File: public/src/components/game.js [hunk 6]
```diff
@@ -32,6 +32,7 @@ export default React.createClass({
 
     return d.div({ className },
       Chat({ hidden: !chat }),
+      d.audio({ id : 'beep', ref: 'beep', src : '/media/beep.wav'}),
```
Comment (aeosynth) [inline public/src/components/game.js]: `ref` is not needed

Comment (aeosynth) [inline public/src/components/game.js]: don't use spaces after a key

File: public/src/cards.js (hunk 4 above)
Comment (sirugh) [inline public/src/cards.js]: Hmm. `self` was defined for me, which is why I opted for that route, but I can switch it back. And the braces are remnants of debugging I forgot to remove.

Comment (aeosynth) [inline public/src/cards.js]: [Window.self](https://developer.mozilla.org/en-US/docs/Web/API/Window.self) is defined, but i'd rather not use it

**Comment by sirugh** [comment]
Haha after looking at this PR, seems like a lot of conversation for two lines of code :P Thanks for your input!

File: public/src/components/game.js [hunk 7]
```diff
@@ -32,6 +32,7 @@ export default React.createClass({
 
     return d.div({ className },
       Chat({ hidden: !chat }),
+      d.audio({id : 'beep' src : '/media/beep.wav'}),
```
Comment (aeosynth) [inline public/src/components/game.js]: `{ id: 'beep', src: '/media/beep.wav' }`

**Comment by aeosynth** [comment]
yeah no worries

**Comment by aeosynth** [comment]
merged: https://github.com/aeosynth/draft/commit/099db9dba1b286bde746d22f5c0722a64641e62f

**Pull request closed without merging**